src/Project/countryDetails.jsx

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useState } from "react";
+import Layout from "./Layout/Layout";
+import { Link, useParams } from "react-router-dom";
+
+
+
+const CountryDetail = () => {
+  const state = useParams();
+  const [countries, setCountries] = useState([]);
+  const [ loading, setLoading ] = useState(true);
+    const fetchCountries = () => {
+        fetch(`https://restcountries.com/v3.1/name/${state.id}/`)
+            .then((res) => {
+              if (res.ok) {
+                    return res.json();
+                }
+            })
+          .then((data) => {
+              setCountries(data);
+              setLoading(false); 
+            })
+            .catch((error) => {
+              console.error('Error fetching data:', error);
+              setLoading(false); 
+            });
+    }
+    useEffect(() => {
+      fetchCountries();
+    }, []);
+  const wait = () => {
```

Comment:
Not an error here. But you can just use the same loading state instead of creating a new function.